src/app/shared/entities/location/location-effects.ts

```diff
@@ -1,28 +1,44 @@
 import { Injectable } from '@angular/core';
 
+import { Store } from '@ngrx/store';
 import { Actions, Effect, ofType} from '@ngrx/effects';
-import { map, tap, switchMap} from 'rxjs/operators';
+import { map, tap, switchMap, take } from 'rxjs/operators';
+import { iif, of } from 'rxjs';
 
 import { LocationApiService } from '@app/api';
 import * as LocationActions from './location-actions';
+import { getLocationsByMap } from './location-selectors';
 
 @Injectable()
 export class LocationEffects {
     constructor(
         private actionStream$: Actions,
-        private locationApi: LocationApiService
+        private locationApi: LocationApiService,
+        private store: Store<any>
     ) {}
 
 
     @Effect()
     fetchMaps$ = this.actionStream$.pipe(
         ofType(LocationActions.fetchAllLocationsByMap),
         switchMap(({map}) => {
-            return this.locationApi.getMapLocations(map.name);
+            // First check the store for locations
+            return this.store.select(getLocationsByMap(map))
```

Comment:
Added check if the store already has the data cached, instead of always hitting the API for the data